frameworks/hdfs/src/main/dist/svc.yml

```diff
@@ -146,16 +156,21 @@ pods:
               ./bootstrap && ./{{HDFS_VERSION}}/bin/hdfs namenode
         resource-set: name-resources
         readiness-check:
-          cmd: export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && ./{{HDFS_VERSION}}/bin/hdfs haadmin -getServiceState name-$POD_INSTANCE_INDEX-node
+          cmd: >
+            export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) &&
+          {{#KERBEROS_ENABLED}}
+            KRB5_CONFIG={{JRE_VERSION}}/lib/security/krb5.conf kinit -k -t $MESOS_SANDBOX/hdfs.keytab $KERBEROS_PRIMARY/$TASK_NAME.$FRAMEWORK_HOST@$KERBEROS_REALM &&
```

Comment:
How do we handle the failure scenarios? For example, a case when for some reason, knit - k -t ... command fails and we do not obtain credentials?

Reply:
When we publish the finalized package, we'll make sure any non-base-tech failure scenarios are covered. Otherwise, any other error will be logged and we can further diagnose based on said logs.

For example, readiness checks will keep retrying and keep failing under a failure scenario, but their failures will be logged, which can then be inspected.